modules/xfem/src/materials/LevelSetBiMaterial.C

```diff
@@ -0,0 +1,76 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "LevelSetBiMaterial.h"
+#include "AuxiliarySystem.h"
+#include "MooseVariable.h"
+#include "XFEM.h"
+
+template <>
+InputParameters
+validParams<LevelSetBiMaterial>()
+{
+  InputParameters params = validParams<Material>();
+  params.addClassDescription(
+      "Compute a material property for bi-materials defined by a level set function.");
+  params.addRequiredParam<NonlinearVariableName>(
+      "level_set_name", "The name of level set variable used to represent the interface");
+  params.addRequiredParam<std::string>("levelset_plus_base",
+                                       "Base name for the material in level set plus domain.");
+  params.addRequiredParam<std::string>("levelset_minus_base",
+                                       "Base name for the material in level set minus domain.");
+  params.addParam<std::string>("base_name",
+                               "Base name for the computed material property (optional)");
+  return params;
+}
+
+LevelSetBiMaterial::LevelSetBiMaterial(const InputParameters & parameters)
+  : Material(parameters),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _ls_var_name(parameters.get<NonlinearVariableName>("level_set_name")),
+    _ls_var(_fe_problem.getStandardVariable(_tid, _ls_var_name)),
+    _aux_system(_fe_problem.getAuxiliarySystem()),
+    _aux_solution(_aux_system.currentSolution()),
+    _use_positive_property(false)
+{
+  FEProblemBase * fe_problem = dynamic_cast<FEProblemBase *>(&_subproblem);
+
+  if (fe_problem == NULL)
+    mooseError("Problem casting _subproblem to FEProblemBase in XFEMMaterialStateMarkerBase");
+  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(fe_problem->getXFEM());
+}
+
+void
+LevelSetBiMaterial::computeProperties()
+{
+  const Node * node = _current_elem->node_ptr(0);
+  dof_id_type ls_dof_id = node->dof_number(_aux_system.number(), _ls_var.number(), 0);
+  Number ls_node_value = (*_aux_solution)(ls_dof_id);
+
+  _use_positive_property = false;
+
+  if (_xfem->isPointInsidePhysicalDomain(_current_elem, *node))
+  {
+    if (ls_node_value > 0.0)
```

Comment:
Is it safe to use the `>` and `<` (below) comparisons here, or would the fuzzy comparisons be appropriate to avoid problems with round off errors?

Reply:
I think it is safe to use > and < here :-)